pkg/unshare/unshare_linux.go

```diff
@@ -76,6 +77,42 @@ func getRootlessGID() int {
 	return os.Getegid()
 }
 
+func isSetUid(path string) (bool, error) {
+	info, err := os.Stat(os.Args[1])
+	if err != nil {
+		return false, err
+	}
+
+	mode := info.Mode()
+	if mode&os.ModeSetuid == os.ModeSetuid {
+		return true, nil
+	}
+
+	cap, err := capability.NewFile(path)
+	if err != nil {
+		return false, err
+	}
+	return cap.Get(capability.EFFECTIVE, capability.CAP_SETUID), nil
+}
+
+func isSetGid(path string) (bool, error) {
```

Comment:
Can you combine isSetUid() and isSetGid() into one function, right now the logic is duplicated, i.e.
```
modeSetid := os.ModeSetuid
capSetid := capability.CAP_SETGID
if gid {
    modeSetid := os.ModeSetuid
    capSetid := capability.CAP_SETGID
}
```